moped-editor/src/components/FeedbackSnackbar.js

```diff
@@ -1,33 +1,87 @@
-import React from "react";
-import { Alert } from '@mui/material';
+import React, { useCallback, useState } from "react";
+import { Alert } from "@mui/material";
 import { Snackbar } from "@mui/material";
 
 /**
- * A project summary snackbar
- * @param {Object} snackbarState - The message you would like to see
- * @param {function} handleSnackbar - The function to update state
- * @param {Number} timeOut - Time out in milliseconds (default: 5000 milliseconds--for five seconds)
+ * Custom hook to manage snackbar state
+ * @returns {Object} - snackbarState, handleSnackbar, handleSnackbarClose
+ */
+export const useFeedbackSnackbar = () => {
+  /**
+   * State for the snackbar
+   * @property {boolean} open - Whether the snackbar is open
+   * @property {String} message - The message to display in the snackbar
+   * @property {String} severity - The MUI Alert severity of the snackbar ("success", "error", "warning", "info")
+   */
+  const [snackbarState, setSnackbarState] = useState({
+    open: false,
+    message: "",
+    severity: "",
+  });
+
+  /**
+   * Wrapper around snackbar state setter
+   * @param {boolean} open - The new state of open
+   * @param {String} message - The message for the snackbar
+   * @param {String} severity - The MUI Alert severity of the snackbar ("success", "error", "warning", "info")
+   * @param {Object} error - The error to be displayed and logged
+   */
+  const handleSnackbar = useCallback(
+    (open, message, severity, error) => {
+      // if there is an error, render error message,
+      // otherwise, render success message
+      if (error) {
+        setSnackbarState({
+          open: open,
+          message: `${message}. Refresh the page to try again.`,
+          severity: severity,
+        });
+        console.error(error);
+      } else {
+        setSnackbarState({
+          open: open,
+          message: message,
+          severity: severity,
+        });
+      }
+    },
+    [setSnackbarState]
+  );
+
+  /**
+   * Callback to reset the snackbar state on snackbar close
+   */
+  const handleSnackbarClose = (_, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setSnackbarState({ open: false, message: "", severity: "" });
+  };
+
+  return { snackbarState, handleSnackbar, handleSnackbarClose };
+};
+
+/**
+ * A feedback summary snackbar to show on success or error events
+ * @param {Object} snackbarState - snackbar state object: { open, message, severity }
+ * @param {function} handleSnackbarClose - Callback function on snackbar close
+ * @param {Number} autoHideDuration - Duration in milliseconds (default: 5000 milliseconds--for five seconds)
  * @returns {JSX.Element}
  * @constructor
  */
 const FeedbackSnackbar = ({
   snackbarState,
-  handleSnackbar,
-  timeOut = 7000,
+  handleSnackbarClose,
+  autoHideDuration = 5000,
```

Comment:
5 seconds feels long enough to me. It also matched what we already had in the docstrings so less edits needed! 😅